src/misc.jl

```diff
@@ -152,40 +175,101 @@ function dijkstra(con_mat::AbstractMatrix{<:Real}, v1)
 end
 
 """
+    floyd_warshall(weights::AbstractMatrix{<:Real})
+
+Compute minimum distances to each node in the network using Floyd-Warshall's algorithm. Here
+we assume that the weigths matrix is symmetric.
+"""
+function floyd_warshall(weights::AbstractMatrix{<:Real})
```

Comment:
http://juliagraphs.github.io/LightGraphs.jl/latest/pathing.html#LightGraphs.floyd_warshall_shortest_paths

Reply:
I think this goes back to that discussion about wether we should add an extra dependency for a single method (that is not something really important in this package). Also, this implementation has a decent performance.

Reply:
Well `dijkstra` was already used prior to this. I think the number of performant algorithms and [integration](http://juliagraphs.github.io/LightGraphs.jl/latest/integration.html) with other packages is a reasonable argument for adding the extra dependency. I'll note that the dependency stack for pandapower is much more complex (e.g., Conda.jl, PyCall.jl, pandapower w/ specific python dependency versions).

Reply:
The pandapower thing is very true. `dijkstra` is not being used anymore, and in this one we actually have comparable performance:
```
julia> using SyntheticGrids
WARNING: `SyntheticGrids.jl may not run properly if inlinning is enabled. Please start Julia with \`--inline=no\``
WARNING: Method definition ==(Base.Nullable{S}, Base.Nullable{T}) in module Base at nullable.jl:238 overwritten in module NullableArrays at /Users/ericperim/.julia/v0.6/NullableArrays/src/operators.jl:99.

julia> grid = Grid(false); nothing

julia> using LightGraphs
WARNING: using LightGraphs.Grid in module Main conflicts with an existing identifier.

julia> graph = LightGraphs.Graph(adjacency(grid))
{137, 167} undirected simple Int64 graph

julia> function mean_short_path_lg(graph::LightGraphs.Graph)
           fws = LightGraphs.floyd_warshall_shortest_paths(graph)
           n = size(fws.dists, 1)
           return sum(fws.dists)/(n * n)
       end
mean_short_path_lg (generic function with 1 method)

julia> for i in 1:1000; el_time[1] += @elapsed SyntheticGrids.mean_shortest_path(adjacency(grid)); el_time[2] += @elapsed mean_short_path_lg(graph); end

julia> println(el_time ./ 1000)
[0.00379599, 0.00483175]
```
of course LightGraphs offers way more functionality, but we are not using them, and that comes at the cost of dealing more types.

Reply:
> cost of dealing more types

By this do you mean that you don't think the 1 extra layer of abstraction is worth it? Wouldn't the `LightGraphs.Graph` type just replace the adjacency matrix?

Reply:
I haven't really looked much into LightGraphs, so maybe my laziness is clouding my judgement here, but I get the impression that several of their methods use custom return types (for instance `floyd_warshall_shortest_paths` returns a `FloydWarshallState` and `dijkstra_shortest_paths` returns a `DijkstraState`), it is not just one extra type (Graph). I am sure that makes sense when you are using further functionalities, but for us here we just really need a simple matrix. On top of that, with our further assumptions we end up with a slightly better performance on this specific task as shown above.
As for the adjacency matrix, it is a necessary piece of the `SyntheticGrids.Grid` object, so it will already be there.

The reason for having pandapower as a part of the stack is that it runs OPF, which is something that we cannot quickly add to the package, while the mean shortest path calculation was simple to add.

Basically, I don't think it is worth integrating with another package just for this one function, but, if you believe we should have all kinds of graph tools at our disposal, then I can go over LightGraphs later and write a proper API to integrate SyntheticGrids with it (for another PR).

Reply:
> for instance floyd_warshall_shortest_paths returns a FloydWarshallState and dijkstra_shortest_paths returns a DijkstraState

FWIW, both those types have a `dists` field that you can extract without really caring about what the type is, but I get your point.

Okay, for now I'd probably suggest opening an issue to integrate with LightGraphs.jl. If we identify more situations that would benefit from using LightGraphs.jl we can keep track of them there.